### PR TITLE
[velero] Add helm annotation for velero.io/v1 schedule CRD

### LINKED ISSUE #151: Helm2 upgrade to v2.12.15 failed
**What steps did you take and what happened:**
[A clear and concise description of what the bug is, and what commands you ran.)
- Create an EKS cluster v1.15
- Install Velero Helm chart
```
helm upgrade --install velero ./velero --namespace velero -f ./values.yaml
```

Helm overriden values.yaml file:
```yaml
configuration:
  provider: aws
  volumeSnapshotLocation:
    config:
      region: eu-central-1
  backupStorageLocation:
    region: eu-central-1
    bucket: velero-backup-xxx
    config:
      region: eu-central-1
initContainers:
  - name: velero-plugin-for-aws
    image: velero/velero-plugin-for-aws
    volumeMounts:
      - mountPath: /target
        name: plugins

schedules:
  backup-k8s:
    schedule: "*/30 * * * *"
    template:
      ttl: "72h"
      includeClusterResources: true
      snapshotVolumes: false
```

**What did you expect to happen:**
Velero helm chart installed successfully

**The output of the following commands will help us better understand what's going on**:
(Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)
```
Release "velero" does not exist. Installing it now.
Error: validation failed: unable to recognize "": no matches for kind "Schedule" in version "velero.io/v1"
```

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- helm version (use `helm version`): v2.16.1
- helm chart version and app version:
  - helm2 (use `helm list`): v2.12.15
  - helm3 (use `helm list -n <YOUR NAMESPACE>`):
- Kubernetes version (use `kubectl version`): v1.15
- Kubernetes installer & version:
- Cloud provider or hardware configuration: EKS
- OS (e.g. from `/etc/os-release`):

### LINKED ISSUE #152: Deployment fails when including schedule in helm
**What steps did you take and what happened:**
When deploying helm (in a rancher k8s cluster) and including a schedule, deployment fails. With the following error:

"Failed to install app velero. Error: validation failed: unable to recognize "": no matches for kind "Schedule" in version "velero.io/v1""

It almost looks like a syntax or parsing error of the values.yaml or schedules.yaml files but I´ve double/triple checked them AND the actual schedule config is copied from an older yaml that works. 
See values yaml at the end (redacted for security and replaced # with && as was screwing with layout in ticket)

Using latest helm charts (2.12.16)

**What did you expect to happen:**
Deploy velero using helm (with restic) and include schedules

**The output of the following commands will help us better understand what's going on**:
(Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- helm version (use `helm version`): version.BuildInfo{Version:"v3.2.0", GitCommit:"e11b7ce3b12db2941e90399e874513fbd24bcb71", GitTreeState:"clean", GoVersion:"go1.13.10"}
- helm chart version and app version:
  - helm2 (use `helm list`):
  - helm3 (use `helm list -n <YOUR NAMESPACE>`): using helm in rancher UI env
- Kubernetes version (use `kubectl version`): v1.18.3
- Kubernetes installer & version: 
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

&&&&
&&&& Configuration settings that directly affect the Velero deployment YAML.
&&&&

&& Details of the container image to use in the Velero deployment & daemonset (if
&& enabling restic). Required.
image:
  repository: dtr01p.abc.com:5000/velero/velero
  tag: v1.4.2_test
  && Digest value example: sha256:d238835e151cec91c6a811fe3a89a66d3231d9f64d09e5f3c49552672d271f38. If used, it will
  && take precedence over the image.tag.
  && digest:
  pullPolicy: IfNotPresent

&& Annotations to add to the Velero deployment's pod template. Optional.
&&
&& If using kube2iam or kiam, use the following annotation with your AWS_ACCOUNT_ID
&& and VELERO_ROLE_NAME filled in:
&&  iam.amazonaws.com/role: arn:aws:iam::<AWS_ACCOUNT_ID>:role/<VELERO_ROLE_NAME>
podAnnotations: {}

&& Additional pod labels for Velero deployment's template. Optional
&& ref: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/
podLabels: {}

&& Resource requests/limits to specify for the Velero deployment. Optional.
resources: {}

&& Init containers to add to the Velero deployment's pod spec. At least one plugin provider image is required.
initContainers:
   - name: velero-plugin-for-aws
     image: dtr01p.abc.com:5000/velero/velero-plugin-for-aws:v1.1.0
     imagePullPolicy: IfNotPresent
     volumeMounts:
       - mountPath: /target
         name: plugins

&& SecurityContext to use for the Velero deployment. Optional.
&& Set fsGroup for `AWS IAM Roles for Service Accounts`
&& see more informations at: https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html
securityContext: {}
  && fsGroup: 1337

&& Tolerations to use for the Velero deployment. Optional.
tolerations: []

&& Affinity to use for the Velero deployment. Optional.
affinity: {}

&& Node selector to use for the Velero deployment. Optional.
nodeSelector: {}

&& Extra volumes for the Velero deployment. Optional.
extraVolumes: []

&& Extra volumeMounts for the Velero deployment. Optional.
extraVolumeMounts: []

&& Settings for Velero's prometheus metrics. Enabled by default.
metrics:
  enabled: true

  && Pod annotations for Prometheus
  podAnnotations:
    prometheus.io/scrape: "true"
    prometheus.io/port: "8085"
    prometheus.io/path: "/metrics"

&& Install CRDs as a templates. Enabled by default.
installCRDs: true

&&&&
&&&& End of deployment-related settings.
&&&&

&&&&
&&&& Parameters for the `default` BackupStorageLocation and VolumeSnapshotLocation,
&&&& and additional server settings.
&&&&
configuration:
  && Cloud provider being used (e.g. aws, azure, gcp).
  provider:

  && Parameters for the `default` BackupStorageLocation. See
  && https://velero.io/docs/v1.4/api-types/backupstoragelocation/
  backupStorageLocation:
    && name is the name of the backup storage location where backups should be stored. If a name is not provided,
    && a backup storage location will be created with the name "default". Optional.
    name:
    && provider is the name for the backup storage location provider. If omitted
    && `configuration.provider` will be used instead.
    provider:
    && bucket is the name of the bucket to store backups in. Required.
    bucket:
    && caCert defines a base64 encoded CA bundle to use when verifying TLS connections to the provider.
    caCert: "********removed*******"
    && prefix is the directory under which all Velero data should be stored within the bucket. Optional.
    prefix:
    && Additional provider-specific configuration. See link above
    && for details of required/optional fields for your provider.
    config:
      region: "us-east-1"
      s3ForcePathStyle: "true"
    &&  s3Url:
    &&  kmsKeyId:
    &&  resourceGroup:
    &&  The ID of the subscription containing the storage account, if different from the cluster’s subscription. (Azure only)
    &&  subscriptionId:
    &&  storageAccount:
    &&  publicUrl:
    &&  Name of the GCP service account to use for this backup storage location. Specify the
    &&  service account here if you want to use workload identity instead of providing the key file.(GCP only)
    &&  serviceAccount:

  && Parameters for the `default` VolumeSnapshotLocation. See
  && https://velero.io/docs/v1.4/api-types/volumesnapshotlocation/
  volumeSnapshotLocation:
    && name is the name of the volume snapshot location where snapshots are being taken. Required.
    name:
    && provider is the name for the volume snapshot provider. If omitted
    && `configuration.provider` will be used instead.
    provider:
    && Additional provider-specific configuration. See link above
    && for details of required/optional fields for your provider.
    config: {}
  &&    region:
  &&    apitimeout:
  &&    resourceGroup:
  &&    The ID of the subscription where volume snapshots should be stored, if different from the cluster’s subscription. If specified, also requires `configuration.volumeSnapshotLocation.config.resourceGroup`to be set. (Azure only)
  &&    subscriptionId:
  &&    snapshotLocation:
  &&    project:

  && These are server-level settings passed as CLI flags to the `velero server` command. Velero
  && uses default values if they're not passed in, so they only need to be explicitly specified
  && here if using a non-default value. The `velero server` default values are shown in the
  && comments below.
  && --------------------
  && `velero server` default: 1m
  backupSyncPeriod:
  && `velero server` default: 1h
  resticTimeout:
  && `velero server` default: namespaces,persistentvolumes,persistentvolumeclaims,secrets,configmaps,serviceaccounts,limitranges,pods
  restoreResourcePriorities:
  && `velero server` default: false
  restoreOnlyMode:

  && additional key/value pairs to be used as environment variables such as "AWS_CLUSTER_NAME: 'yourcluster.domain.tld'"
  extraEnvVars: {}

  && Comma separated list of velero feature flags. default: empty
  features:

  && Set log-level for Velero pod. Default: info. Other options: debug, warning, error, fatal, panic.
  logLevel:

  && Set log-format for Velero pod. Default: text. Other option: json.
  logFormat:

&&&&
&&&& End of backup/snapshot location settings.
&&&&

&&&&
&&&& Settings for additional Velero resources.
&&&&

rbac:
  && Whether to create the Velero role and role binding to give all permissions to the namespace to Velero.
  create: true
  && Whether to create the cluster role binding to give administrator permissions to Velero
  clusterAdministrator: true

&& Information about the Kubernetes service account Velero uses.
serviceAccount:
  server:
    create: true
    name:
    annotations:

&& Info about the secret to be used by the Velero deployment, which
&& should contain credentials for the cloud provider IAM account you've
&& set up for Velero.
credentials:
  && Whether a secret should be used as the source of IAM account
  && credentials. Set to false if, for example, using kube2iam or
  && kiam to provide IAM credentials for the Velero pod.
  useSecret: true
  && Name of a pre-existing secret (if any) in the Velero namespace
  && that should be used to get IAM account credentials. Optional.
  existingSecret:
  && Data to be stored in the Velero secret, if `useSecret` is
  && true and `existingSecret` is empty. This should be the contents
  && of your IAM credentials file.
  secretContents: {}
  && additional key/value pairs to be used as environment variables such as "DIGITALOCEAN_TOKEN: <your-key>". Values will be stored in the secret.
  extraEnvVars: {}

&& Whether to create backupstoragelocation crd, if false => do not create a default backup location
backupsEnabled: true
&& Whether to create volumesnapshotlocation crd, if false => disable snapshot feature
snapshotsEnabled: true

&& Whether to deploy the restic daemonset.
deployRestic: true

restic:
  podVolumePath: /var/lib/kubelet/pods
  privileged: false
  && Pod priority class name to use for the Restic daemonset. Optional.
  priorityClassName: {}
  && Resource requests/limits to specify for the Restic daemonset deployment. Optional.
  resources: {}
  && Tolerations to use for the Restic daemonset. Optional.
  tolerations: []

  && Extra volumes for the Restic daemonset. Optional.
  extraVolumes: []

  && Extra volumeMounts for the Restic daemonset. Optional.
  extraVolumeMounts: []

  && SecurityContext to use for the Velero deployment. Optional.
  && Set fsGroup for `AWS IAM Roles for Service Accounts`
  && see more informations at: https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html
  securityContext: {}
    && fsGroup: 1337

&& Backup schedules to create.
&& Eg:
&& schedules:
&&   mybackup:
&&      labels:
&&        myenv: foo
&&     schedule: "0 0 * * *"
&&     template:
&&       ttl: "240h"
&&       includedNamespaces:
&&        - foo

schedules:
  hourly:
    schedule: "0 6,10,14,18,22 * * *"
    template:
      ttl: "168h"
  daily:
    schedule: "0 0 * * *"
    template:
      ttl: "720h"

&&schedules: {}
&& Velero ConfigMaps.
&& Eg:
configMaps:
  restic-restore-action-config:
    labels:
      velero.io/plugin-config: ""
      velero.io/restic: RestoreItemAction
    data:
      image: dtr01p.abc.com:5000/velero/velero-restic-restore-helper:v1.4.2
&&configMaps: {}

&&&&
&&&& End of additional Velero resource settings.
&&&&

### LINKED ISSUE #160: Schedules are not created with Helm 2.16.1
**What steps did you take and what happened:**
- Installing via Helm 2.16.1 via tiller into EKS k8s 1.15.11
- Issued two schedules via the helm values, one with EBS snapshot and one without
```yaml
schedules:
  nightly:
    schedule: "17 2 * * *"
    template:
      ttl: "336h"
  nightly-exclude-snapshot:
    schedule: "27 2 * * *"
    template:
      snapshotVolumes: false
      ttl: "336h"
```
- Installation of the chart was green, but the `schedule` resources were missing
```
helm upgrade --install --namespace velero -f values.velero.yaml velero vmware-tanzu/velero

...

==> MISSING
KIND                                NAME
velero.io/v1, Resource=schedules    velero-nightly-exclude-snapshot
velero.io/v1, Resource=schedules    velero-nightly

```
- Confirmed missing `schedule` resources
```
kubectl get schedules.velero.io -A                                                                                                                                             
No resources found
```

**What did you expect to happen:**
`schedule` resources to be created.

**Anything else you would like to add:**
I downloaded the same version of the Velero Helm Chart and templated it with my values files, which generated the `schedule` manifests
```yaml
---
# Source: velero/templates/schedule.yaml

apiVersion: velero.io/v1
kind: Schedule
metadata:
  name: release-name-velero-nightly
  labels:
    app.kubernetes.io/name: velero
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/managed-by: Tiller
    helm.sh/chart: velero-2.12.17
spec:
  schedule: "17 2 * * *"
  template:
    ttl: 336h

---
apiVersion: velero.io/v1
kind: Schedule
metadata:
  name: release-name-velero-nightly-exclude-snapshot
  labels:
    app.kubernetes.io/name: velero
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/managed-by: Tiller
    helm.sh/chart: velero-2.12.17
spec:
  schedule: "27 2 * * *"
  template:
    snapshotVolumes: false
    ttl: 336h

---
```
I then applied this manually against the cluster and got a success return
```bash
kubectl apply -f velero-schedule.yaml 
schedule.velero.io/release-name-velero-nightly created
schedule.velero.io/release-name-velero-nightly-exclude-snapshot created

kubectl get schedules.velero.io                                                                                                                                               
NAME                                           AGE
release-name-velero-nightly                    36s
release-name-velero-nightly-exclude-snapshot   36s
```

**Environment:**

- helm version (use `helm version`): 2.16.1
- helm chart version and app version: 
  - Chart - velero-2.12.17
  - App - 1.4.2
- Kubernetes version (use `kubectl version`): 1.15.11
- Cloud provider or hardware configuration: AWS EKS

### LINKED ISSUE #165: Schedule CRD got removed after Schedule template created on helm-2.16.10
**What steps did you take and what happened:**
enabled `installCRDs: true` flag after upgrade helm-2.12.x to helm-2.16.10, and during `helm upgrade --install` step getting schedule CRD deleted after schedule template being created, which lead to removing schedule template.
As I understand, it because of post-upgrade hook in CRD
```
---
apiVersion: apiextensions.k8s.io/v1beta1
kind: CustomResourceDefinition
metadata:
  labels:
    app.kubernetes.io/name: "velero"
  annotations:
    "helm.sh/hook": pre-install,post-upgrade
    "helm.sh/hook-delete-policy": "before-hook-creation"
    controller-gen.kubebuilder.io/version: v0.2.4
  creationTimestamp: null
  name: schedules.velero.io
```

**What did you expect to happen:**

get Schedule template not deleted after install

**Anything else you would like to add:**
Here is part of tiller log which shows that we install schedule template first, and delete/install CRD after that
```
[kube] 2020/09/20 08:45:13 checking 6 resources for changes
[kube] 2020/09/20 08:45:13 Looks like there are no changes for ServiceAccount "velero-server" in "velero"
[kube] 2020/09/20 08:45:13 Looks like there are no changes for ClusterRoleBinding "velero-server" in ""
[kube] 2020/09/20 08:45:13 Looks like there are no changes for Role "velero-server" in "velero"
[kube] 2020/09/20 08:45:13 Looks like there are no changes for RoleBinding "velero-server" in "velero"
[kube] 2020/09/20 08:45:13 Looks like there are no changes for Deployment "velero" in "velero"
add ---> [kube] 2020/09/20 08:45:13 Created a new Schedule called "velero-daily" in velero
[kube] 2020/09/20 08:45:13 beginning wait for 6 resources with timeout of 5m0s
[tiller] 2020/09/20 08:45:16 executing 13 post-upgrade hooks for velero
[tiller] 2020/09/20 08:45:16 deleting post-upgrade hook default for release velero due to "before-hook-creation" policy
[kube] 2020/09/20 08:45:17 Starting delete for "default" BackupStorageLocation
[kube] 2020/09/20 08:45:17 backupstoragelocations.velero.io "default" not found
[kube] 2020/09/20 08:45:17 Waiting for 60 seconds for delete to be completed
[kube] 2020/09/20 08:45:19 building resources from manifest
[kube] 2020/09/20 08:45:19 creating 1 resource(s)
[kube] 2020/09/20 08:45:19 Watching for changes to BackupStorageLocation default in velero with timeout of 5m0s
[kube] 2020/09/20 08:45:19 Add/Modify event for default: ADDED
[tiller] 2020/09/20 08:45:19 deleting post-upgrade hook gcp for release velero due to "before-hook-creation" policy
[kube] 2020/09/20 08:45:19 Starting delete for "gcp" VolumeSnapshotLocation
[kube] 2020/09/20 08:45:19 volumesnapshotlocations.velero.io "gcp" not found
[kube] 2020/09/20 08:45:19 Waiting for 60 seconds for delete to be completed
[kube] 2020/09/20 08:45:21 building resources from manifest
[kube] 2020/09/20 08:45:22 creating 1 resource(s)
[kube] 2020/09/20 08:45:22 Watching for changes to VolumeSnapshotLocation gcp in velero with timeout of 5m0s
[kube] 2020/09/20 08:45:22 Add/Modify event for gcp: ADDED
[tiller] 2020/09/20 08:45:22 deleting post-upgrade hook schedules.velero.io for release velero due to "before-hook-creation" policy
delete ---> [kube] 2020/09/20 08:45:23 Starting delete for "schedules.velero.io" CustomResourceDefinition
[kube] 2020/09/20 08:45:23 Waiting for 60 seconds for delete to be completed
[kube] 2020/09/20 08:45:25 building resources from manifest
[kube] 2020/09/20 08:45:25 creating 1 resource(s)
[kube] 2020/09/20 08:45:25 Watching for changes to CustomResourceDefinition schedules.velero.io in  with timeout of 5m0s
[kube] 2020/09/20 08:45:25 Add/Modify event for schedules.velero.io: ADDED
[tiller] 2020/09/20 08:45:25 hooks complete for post-upgrade velero
[storage] 2020/09/20 08:45:25 updating release "velero.v254"
```

**Environment:**

- helm version: 2.16.10
- helm chart version and app version:
  chart: 2.12.13
  velero app: 1.4.2
- Kubernetes version:
```
Server Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.12-gke.20", GitCommit:"0ac5f81eecab42bff5ef74f18b99d8896ba7b89b", GitTreeState:"clean", BuildDate:"2020-09-09T00:48:20Z", GoVersion:"go1.12.17b4", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration: GKE
- OS (e.g. from `/etc/os-release`): 
```
BUILD_ID=12371.1079.0
NAME="Container-Optimized OS"
KERNEL_COMMIT_ID=52bdab9330bdd9e50dc967f8aa850829921ca217
GOOGLE_CRASH_ID=Lakitu
VERSION_ID=77
BUG_REPORT_URL="https://cloud.google.com/container-optimized-os/docs/resources/support-policy#contact_us"
PRETTY_NAME="Container-Optimized OS from Google"
VERSION=77
GOOGLE_METRICS_PRODUCT_ID=26
HOME_URL="https://cloud.google.com/container-optimized-os/docs"
ID=cos
```

### LINKED ISSUE #167: [bug] First install CRD's, then install resources that depends on CRD's
**What steps did you take and what happened:**
[A clear and concise description of what the bug is, and what commands you ran.)

1. Try to up Velero in new cluster using [this `values.yaml`](https://github.com/MaxymVlasov/helm-charts-1/blob/8a653ce4750d0ca7b579467f301192c04b48b17d/charts/velero/examples/vsphere/values.yaml) from #166 
2. Run `helmfile -f helmfile.yaml diff` and `helmfile -f helmfile.yaml apply`.
3. Get same error `... no matches for kind "Schedule" in version "velero.io/v1"` - it means that CRD resources not yet exist.

**What did you expect to happen:**
Schedules should be created after created CRD resources. At least `apply` should be done without errors

**The output of the following commands will help us better understand what's going on**:
https://pastebin.com/EeZ2KPWX

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]
Could be related to: #44 

**Environment:**

- helm version (use `helm version`): v3.3.1"
- helm chart version and app version: 
  - Chart: 2.12.17
  - app: v1.4.2
- Kubernetes version (use `kubectl version`): v1.17.2
- Kubernetes installer & version: Rancher v2.3.5
- Cloud provider or hardware configuration: vSphere
- OS (e.g. from `/etc/os-release`): Ubuntu 18.04.3 LTS

### PR DESCRIPTION
**Pull request by jenting**
#### Special notes for your reviewer:

fixes #151
fixes #152
fixes #160
fixes #165
fixes #167

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[velero]`)

### CODE REVIEW

**Comment by jenting** [comment]
@namm2, @reddiablo85 PTAL :bow:

**Comment by namm2** [comment]
I tested and it works fine. Thanks for the PR.

**Review by carlisia — DISMISSED** [review]
👍  thank you for this fix!

**Comment by carlisia** [comment]
@jenting this will need a version bump please.

**Comment by jenting** [comment]
@carlisia @ashish-amarnath @nrb @cpanato could you please take a look when you're available? thanks.

**Comment by jenting** [comment]
@carlisia @ashish-amarnath @nrb @cpanato PTAL 🙇

File: charts/velero/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 appVersion: 1.4.2
 description: A Helm chart for velero
 name: velero
-version: 2.12.17
+version: 2.12.18
```
Comment (cpanato) [inline charts/velero/Chart.yaml:5]: this looks like a new fetaur I would bump to 2.13.0

Comment (jenting) [reply charts/velero/Chart.yaml:5]: It's a bug fixing, lots of issues have been reported recently and I think we missed the annotation to `kind: Schedule` at PR #109.

Comment (namm2) [reply charts/velero/Chart.yaml:5]: This is just a bug fix. I hope it could be merged as soon as possible.

**Review by cpanato — DISMISSED** [review]

File: charts/velero/Chart.yaml (hunk 1 above)
Comment (Vladikamira) [reply charts/velero/Chart.yaml:5]: also would like to see it merged 👍 🙏

Comment (jenting) [reply charts/velero/Chart.yaml]: @cpanato could you please take a look at this PR, thanks :bow:

**Review by cpanato — APPROVED** [review]
thanks for this!
lgtm

**Review by carlisia — APPROVED** [review]
👍  thank you!

**Pull request merged**